### PR TITLE
Fix/sankey gap chart

### PR DESCRIPTION
**Pull request by xile611**
[[中文版模板 / Chinese template](https://github.com/VisActor/vgrammar/blob/main/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Refactoring
- [ ] Update dependency
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Workflow
- [ ] Chore
- [ ] Release
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### 🚀 Summary

copilot:summary

### 🔍 Walkthrough

copilot:walkthrough

### CODE REVIEW

File: packages/vgrammar-sankey/src/layout.ts [hunk 1]
```diff
@@ -633,22 +633,25 @@ export class SankeyLayout {
...
-        minNodeHeight > 0 ? Math.max(this.options.nodeGap, minNodeHeight) : this.options.nodeGap,
-        this._viewBox.height / maxRowCount
-      );
+      const maxStepHeight = this._viewBox.height / maxRowCount;
+      const gapY = Math.min(this.options.nodeGap, maxStepHeight);
       getGapY = () => gapY;
       this._gapY = gapY;
 
+      if ((minNodeHeight + gapY) * maxRowCount > this._viewBox.height) {
+        minNodeHeight = maxStepHeight - gapY;
```
Comment (xiaoluoHe) [inline packages/vgrammar-sankey/src/layout.ts:642]: 这里 maxStepHeight - gapY 看着有可能小于 0，要处理一下吗？

Comment (xiaoluoHe) [reply packages/vgrammar-sankey/src/layout.ts:642]: 看到下面 L648 的判断了～

**Review by xiaoluoHe — APPROVED** [review]

**Comment by xiaoluoHe** [comment]
补充下 bugserver case?

**Comment by xile611** [comment]
@xiaoluoHe 已补充
669a361ecede4100c6d5ded3
669a3a753fdce800ccf994fe

**Pull request merged**